service/src/components/carousel/Carousel.tsx

```diff
@@ -0,0 +1,129 @@
+import { useState, useRef, MouseEvent, useCallback } from 'react';
+import { RiArrowDropLeftLine, RiArrowDropRightLine } from 'react-icons/ri';
+
+import { CarouselProps } from './type';
+
+export default function Carousel({ data }: CarouselProps) {
+    const [current, setCurrent] = useState(0);
+    const [isDrag, setIsDrag] = useState(false);
+    const [startX, setStartX] = useState<number>(0);
+    const scrollRef = useRef<HTMLDivElement | null>(null);
+
+    const prevSlider = () => {
+        setCurrent(current === 0 ? data.length - 1 : current - 1);
+    };
+    const nextSlider = () => {
+        setCurrent(current === data.length - 1 ? 0 : current + 1);
+    };
+
+    const onDragStart = (e: MouseEvent<HTMLElement>) => {
+        e.preventDefault();
+        setIsDrag(true);
+        if (scrollRef.current) {
+            setStartX(e.pageX + scrollRef.current.scrollLeft);
+        }
+    };
+    const onDragEnd = () => {
+        setIsDrag(false);
+    };
+
+    const onDragMove = useCallback(
+        (e: MouseEvent<HTMLButtonElement>) => {
+            e.preventDefault();
+            if (isDrag) {
+                const { scrollWidth, clientWidth, scrollLeft }: any = scrollRef.current;
+                if (scrollRef.current) {
+                    scrollRef.current.scrollLeft = startX - e.pageX;
+                }
+
+                if (scrollLeft === 0) {
+                    setStartX(e.pageX); //가장 왼쪽일 때, 움직이고 있는 마우스의 x좌표가 곧 startX로 설정.
+                } else if (scrollWidth <= clientWidth + scrollLeft) {
+                    setStartX(e.pageX + scrollLeft); //가장 오른쪽일 때, 움직이고 있는 마우스의 x좌표에 현재 스크롤된 길이 scrollLeft의 합으로 설정
+                }
+            }
+        },
+        [isDrag, startX],
+    );
+
+    // 쓰로틀 구현
+    const throttle = (func: any, ms: number) => {
```

Comment:
func: any보다는 아래처럼 바꾸는 것은 어떤가요!
 any 보다는 함수임을 나타내면 좋을 것 같습니다
```suggestion
    const throttle = (func: (...args: any[])=>void, ms: number) => {
```